modules/tinymce/src/core/main/ts/annotate/AnnotationFilter.ts

```diff
@@ -6,17 +6,42 @@ import { AnnotationsRegistry, AnnotatorSettings } from './AnnotationsRegistry';
 import * as Markings from './Markings';
 
 const setup = (editor: Editor, registry: AnnotationsRegistry): void => {
-  const identifyParserNode = (span: AstNode): Optional<AnnotatorSettings> => Optional.from(span.attr(Markings.dataAnnotation())).bind(registry.lookup);
+  const dataAnnotation = Markings.dataAnnotation();
+  const identifyParserNode = (node: AstNode): Optional<AnnotatorSettings> =>
+    Optional.from(node.attr(dataAnnotation)).bind(registry.lookup);
+
+  const removeDirectAnnotation = (node: AstNode) => {
+    node.attr(Markings.dataAnnotationId(), null);
+    node.attr(Markings.dataAnnotation(), null);
+    node.attr(Markings.dataAnnotationActive(), null);
+
+    const customAttrNames = Optional.from(node.attr(Markings.dataAnnotationAttributes())).map((names) => names.split(',')).getOr([]);
+    const customClasses = Optional.from(node.attr(Markings.dataAnnotationClasses())).map((names) => names.split(',')).getOr([]);
+    Arr.each(customAttrNames, (name) => node.attr(name, null));
+
+    const classList = new Set(node.attr('class')?.split(' ') ?? []);
+    Arr.each([ Markings.annotation() ].concat(customClasses), (clazz) => classList.delete(clazz));
+    const newClassList = Array.from(classList.values());
```

Comment:
This can be simplified to this I believe?
```suggestion
    const classList = node.attr('class')?.split(' ') ?? [];
    const newClassList = Arr.difference(classList, [ Markings.annotation() ].concat(customClasses));
```

Reply:
Yep, that should work. Thanks